core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -303,11 +303,15 @@ class AdminManager(val config: KafkaConfig,
 
           case ResourceType.TOPIC =>
             val topic = resource.name
-            Topic.validate(topic)
-            // Consider optimizing this by caching the configs or retrieving them from the `Log` when possible
-            val topicProps = adminZkClient.fetchEntityConfig(ConfigType.Topic, topic)
-            val logConfig = LogConfig.fromProps(KafkaServer.copyKafkaConfigToLog(config), topicProps)
-            createResponseConfig(allConfigs(logConfig), createTopicConfigEntry(logConfig, topicProps, includeSynonyms))
+            if (metadataCache.contains(topic)) {
+              Topic.validate(topic)
```

Comment:
Should the topic validation go first? Presumably, if it's in the cache, it must be valid?

Reply:
@hachikuji  Thanks for the review. yes, topic validation should come first. updated the code.